CI failure: Deploy — credential files contain literal placeholders instead of actual values

## CI Failure Report

**Workflow:** Deploy
**Branch:** main
**Commit:** 9c44a757b37d0b707a58f5ead61d06dbc22f9246
**Run:** https://github.com/Action-Llama/agents/actions/runs/23414371939

## Failure Summary

The deploy workflow is failing because the `al push` command cannot find valid credentials. The credential JSON files are being written with literal GitHub Actions placeholder strings instead of the actual secret values due to incorrect shell quoting.

## Failed Job(s)

### deploy

**Error output:**
```
  [MISSING] GitHub Personal Access Token (github_token)
Credential error: 3 credential(s) missing: github_token, git_ssh, anthropic_key.
Run 'al doctor' interactively to configure them.
  [MISSING] SSH Key & Git Identity (git_ssh)
  [MISSING] Anthropic API Credential (anthropic_key)
##[error]Process completed with exit code 1.
```

## Root Cause Analysis

The issue is in the "Configure credentials for action-llama" step in `.github/workflows/deploy.yml` (lines 41-55). The credential JSON files are being written using single quotes around the entire JSON string:

```bash
echo '{"type":"github_token","token":"${{ secrets.GITHUB_TOKEN }}"}' > ~/.action-llama/credentials/github_token.json
```

In bash, single quotes prevent all variable expansion and substitution. This means the literal string `${{ secrets.GITHUB_TOKEN }}` is being written to the file instead of the actual token value. The same issue affects all three credential files (github_token.json, git_ssh.json, and anthropic_key.json).

## Suggested Fix

Replace the single quotes with double quotes and escape the JSON quotes properly. Here's the fix for the "Configure credentials for action-llama" step:

```diff
      - name: Configure credentials for action-llama
        run: |
          # Set up the credential directory
          mkdir -p ~/.action-llama/credentials
          
          # Configure GitHub token
-         echo '{"type":"github_token","token":"${{ secrets.GITHUB_TOKEN }}"}' > ~/.action-llama/credentials/github_token.json
+         echo "{\"type\":\"github_token\",\"token\":\"${{ secrets.GITHUB_TOKEN }}\"}" > ~/.action-llama/credentials/github_token.json
          
          # Configure SSH/Git credentials - use the deploy key  
          GIT_EMAIL="${{ vars.GIT_EMAIL }}"
          GIT_NAME="${{ vars.GIT_NAME }}"
-         echo '{"type":"git_ssh","privateKey":"${{ secrets.DEPLOY_SSH_KEY }}","email":"'${GIT_EMAIL:-deploy@action-llama.com}'","name":"'${GIT_NAME:-Action Llama Deploy}'"}' > ~/.action-llama/credentials/git_ssh.json
+         echo "{\"type\":\"git_ssh\",\"privateKey\":\"${{ secrets.DEPLOY_SSH_KEY }}\",\"email\":\"${GIT_EMAIL:-deploy@action-llama.com}\",\"name\":\"${GIT_NAME:-Action Llama Deploy}\"}" > ~/.action-llama/credentials/git_ssh.json
          
          # Configure Anthropic API key 
-         echo '{"type":"anthropic_key","key":"${{ secrets.ANTHROPIC_API_KEY }}"}' > ~/.action-llama/credentials/anthropic_key.json
+         echo "{\"type\":\"anthropic_key\",\"key\":\"${{ secrets.ANTHROPIC_API_KEY }}\"}" > ~/.action-llama/credentials/anthropic_key.json
```

Alternatively, use printf for better readability:
```bash
printf '{"type":"github_token","token":"%s"}' "${{ secrets.GITHUB_TOKEN }}" > ~/.action-llama/credentials/github_token.json
```

## Additional Context

This issue was introduced after removing the `--no-creds` flag in commit 9c44a75. The workflow was previously skipping credential validation, but now requires properly formatted credential files. The commit history shows multiple recent attempts to fix CI deployment issues, suggesting this has been an ongoing problem.

---
*This issue was automatically created by the gh-actions-responder agent.*

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -43,15 +43,15 @@ jobs:
           mkdir -p ~/.action-llama/credentials
           
           # Configure GitHub token
-          echo '{"type":"github_token","token":"${{ secrets.GITHUB_TOKEN }}"}' > ~/.action-llama/credentials/github_token.json
+          printf '{"type":"github_token","token":"%s"}\n' "${{ secrets.GITHUB_TOKEN }}" > ~/.action-llama/credentials/github_token.json
           
           # Configure SSH/Git credentials - use the deploy key  
           GIT_EMAIL="${{ vars.GIT_EMAIL }}"
           GIT_NAME="${{ vars.GIT_NAME }}"
-          echo '{"type":"git_ssh","privateKey":"${{ secrets.DEPLOY_SSH_KEY }}","email":"'${GIT_EMAIL:-deploy@action-llama.com}'","name":"'${GIT_NAME:-Action Llama Deploy}'"}' > ~/.action-llama/credentials/git_ssh.json
+          printf '{"type":"git_ssh","privateKey":"%s","email":"%s","name":"%s"}\n' "${{ secrets.DEPLOY_SSH_KEY }}" "${GIT_EMAIL:-deploy@action-llama.com}" "${GIT_NAME:-Action Llama Deploy}" > ~/.action-llama/credentials/git_ssh.json
           
           # Configure Anthropic API key 
-          echo '{"type":"anthropic_key","key":"${{ secrets.ANTHROPIC_API_KEY }}"}' > ~/.action-llama/credentials/anthropic_key.json
+          printf '{"type":"anthropic_key","key":"%s"}\n' "${{ secrets.ANTHROPIC_API_KEY }}" > ~/.action-llama/credentials/anthropic_key.json
 
       - name: Deploy
         env:

